cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java

```diff
@@ -137,11 +139,16 @@ public void getEndpointSummary(HttpRequest request, HttpResponder responder,
                                  @PathParam("program-type") String programType,
                                  @PathParam("program-name") String programName,
                                  @PathParam("run-id") String runId) throws Exception {
-    ProgramReference programReference = new ProgramReference(namespaceId, appName,
-                                                             ProgramType.valueOfCategoryName(programType),
-                                                             programName);
-    EndpointsSummary endpointSummary = fieldLineageAdmin.getEndpoints(namespaceId, programReference, runId);
-    responder.sendString(HttpResponseStatus.OK, GSON.toJson(endpointSummary));
+    try {
+      RunId run = RunIds.fromString(runId);
+      ProgramReference programReference = new ProgramReference(namespaceId, appName,
+                                                               ProgramType.valueOfCategoryName(programType),
+                                                               programName);
+      EndpointsSummary endpointSummary = fieldLineageAdmin.getEndpoints(namespaceId, programReference, run);
+      responder.sendString(HttpResponseStatus.OK, GSON.toJson(endpointSummary));
+    } catch (Exception e) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, e.getMessage());
```

Comment:
this is correct if there is an exception from the RunId and ProgramReference creation, but not for the getEndpoints() call. It's better to move that call outside of the try-catch